handler/api/src/main/java/io/knotx/fragments/handler/api/metadata/NodeMetadata.java

```diff
@@ -78,6 +92,44 @@ public OperationMetadata getOperation() {
     return operation;
   }
 
+  public void markProcessingStart() {
+    processingStartTimestamp = System.currentTimeMillis();
+  }
+
+  public void markProcessingEnd() {
+    processingEndTimestamp = System.currentTimeMillis();
+  }
+
+  public void calculateTimestampsBasedOnSubtasks(Map<String, NodeMetadata> nodes) {
+    if (type == NodeType.COMPOSITE) {
+      List<NodeMetadata> nestedMetadata = nestedNodes.stream()
+          .map(nodes::get)
+          .collect(Collectors.toList());
+
+      nestedMetadata.forEach(nodeMetadata -> nodeMetadata.calculateTimestampsBasedOnSubtasks(nodes));
+
+      processingStartTimestamp = nestedMetadata.stream()
+          .map(NodeMetadata::getProcessingStartTimestamp)
+          .min(Comparator.naturalOrder())
+          .orElse((long) 0);
+
+      processingEndTimestamp = getLatestNestedTime(nodes);
+    }
+  }
+
+  private long getLatestNestedTime(Map<String, NodeMetadata> nodes) {
```

Comment:
This logic is invalid. The lastestNestedTime is not based on subtasks, but on the transitions of this CompositeNode - please fix

Reply:
Right, good catch

Reply:
Actually, when I'm looking at it now, this logic is correct. It has wrong name though. Subtasks are passed into this method, and it looks for the deepest nodes in them

Reply:
I still disagree. This method calculates the latest completion time of all the nodes reachable via transistions from the current node - never diving into subtasks for composite nodes.
And, by the way, there should only be a single transition to traverse at each step by the way (unless we're preparing for some substantial change in processing logic?)

Reply:
Yup, fixed